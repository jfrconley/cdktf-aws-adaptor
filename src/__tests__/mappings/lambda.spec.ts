import { LambdaLayerVersionPermission } from "@cdktf/provider-aws/lib/lambda-layer-version-permission/index.js";
import { LambdaLayerVersion } from "@cdktf/provider-aws/lib/lambda-layer-version/index.js";
import { LambdaPermission } from "@cdktf/provider-aws/lib/lambda-permission/index.js";
import { CfnLayerVersion, CfnLayerVersionPermission, CfnPermission } from "aws-cdk-lib/aws-lambda";
import { describe } from "vitest";
import { itShouldMapCfnElementToTerraformResource } from "../helpers.js";

describe("Lambda mappings", () => {
  itShouldMapCfnElementToTerraformResource(
    CfnPermission,
    {
      action: "lambda:InvokeFunction",
      functionName: "my-function",
      principal: "sns.amazonaws.com",
      sourceArn: "arn:aws:sns:us-east-1:123456789012:my-topic-1a2b3c4d",
      eventSourceToken: "token",
      functionUrlAuthType: "IAM",
      principalOrgId: "123456789012",
      sourceAccount: "123456789012",
    },
    LambdaPermission,
    {
      action: "lambda:InvokeFunction",
      functionName: "my-function",
      principal: "sns.amazonaws.com",
      sourceArn: "arn:aws:sns:us-east-1:123456789012:my-topic-1a2b3c4d",
      eventSourceToken: "token",
      functionUrlAuthType: "IAM",
      principalOrgId: "123456789012",
      sourceAccount: "123456789012",
    },
  );

  itShouldMapCfnElementToTerraformResource(
    CfnLayerVersion,
    {
      compatibleArchitectures: ["x86_64"],
      compatibleRuntimes: ["nodejs12.x"],
      content: {
        s3Bucket: "my-bucket",
        s3Key: "my-key",
        s3ObjectVersion: "my-version",
      },
      description: "my description",
      layerName: "my-layer",
      licenseInfo: "my-license",
    },
    LambdaLayerVersion,
    {
      s3Bucket: "my-bucket",
      s3Key: "my-key",
      s3ObjectVersion: "my-version",
      compatibleArchitectures: ["x86_64"],
      compatibleRuntimes: ["nodejs12.x"],
      description: "my description",
      layerName: "my-layer",
      licenseInfo: "my-license",
    },
  );

  itShouldMapCfnElementToTerraformResource(
    CfnLayerVersionPermission,
    {
      action: "lambda:GetLayerVersion",
      principal: "123456789012",
      organizationId: "o-123456",
      layerVersionArn: "arn:aws:lambda:us-east-1:123456789012:layer:my-layer:1",
    },
    LambdaLayerVersionPermission,
    {
      action: "lambda:GetLayerVersion",
      principal: "123456789012",
      organizationId: "o-123456",
      layerName: "arn:aws:lambda:us-east-1:123456789012:layer:my-layer:1",
      versionNumber:
        `$\{tonumber(element(split(\":\", \"arn:aws:lambda:us-east-1:123456789012:layer:my-layer:1\"), 7))}` as unknown as number,
      statementId: "teststackawstackresourceC278F851",
    },
  );
});
