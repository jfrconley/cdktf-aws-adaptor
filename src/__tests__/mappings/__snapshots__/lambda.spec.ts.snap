// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Lambda mappings > should map CfnAlias to LambdaAlias 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_lambda_alias\\": {
      \\"resource_22C949BF\\": {
        \\"description\\": \\"description\\",
        \\"function_name\\": \\"functionName\\",
        \\"function_version\\": \\"functionVersion\\",
        \\"name\\": \\"name\\",
        \\"routing_config\\": {
          \\"additional_version_weights\\": {
            \\"1\\": 2,
            \\"2\\": 1
          }
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.23.1\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_lambda_alias.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
